protocol/flows/src/flow_context.rs

```diff
@@ -397,6 +397,22 @@ impl FlowContext {
             });
         }
 
+        // TODO: refactor this adding a worker and a scheduler to FlowContext
+        if self.should_expire_transactions().await {
+            // Spawn a task expiring concurrently the low priority transactions.
+            // The TransactionSpread instance ensures at most one expire running at any
+            // given time.
+            let mining_manager = self.mining_manager().clone();
+            let consensus_clone = consensus.clone();
+            let context = self.clone();
+            tokio::spawn(async move {
+                tokio::spawn(async move {
+                    mining_manager.expire_low_priority_transactions(&consensus_clone).await;
+                });
+                context.expire_done().await;
```

Comment:
Due to the double `tokio::spawn` this might run before `expire_low_priority_transactions` executes. Is that the intended behavior?